hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/RationalizedSigMeta.java

```diff
@@ -138,6 +145,40 @@ public void revokeCryptoSigsFrom(final JKey key) {
                                 : wrappedFn.apply(publicKey);
     }
 
+    public void replacePayerHollowKeyIfNeeded(SignatureMap signatureMap) {
+        if (!payerReqSig.hasHollowKey() || rationalizedSigs == null) return;
+
+        final var targetEvmAddress = payerReqSig.getHollowKey().getEvmAddress();
+        for (final var sig : rationalizedSigs) {
+            // maybe do the hashing of the public key a better way... not coupling to Besu classes?
+            final var publicKeyHashed =
+                    Hash.hash(Bytes.of(sig.getExpandedPublicKey())).toArrayUnsafe();
+            if (Arrays.equals(
+                    targetEvmAddress,
+                    0,
+                    targetEvmAddress.length,
+                    publicKeyHashed,
+                    publicKeyHashed.length - 20,
+                    publicKeyHashed.length)) {
+
+                // use compressed public key
```

Comment:
As discussed, compressed key is just `<parity byte of s-coordinate> | r-coordinate`.

Reply:
Added utility do derive the compressed key from the uncompressed one.